Sources/DistributedActors/Serialization/Serialization+Invocation.swift

```diff
@@ -17,22 +17,26 @@ import Logging
 import NIO
 import NIOFoundationCompat
 import SwiftProtobuf
-import SWIM
-
 import Foundation // for Codable
 
 public struct ClusterInvocationEncoder: DistributedTargetInvocationEncoder {
     public typealias SerializationRequirement = any Codable
     var arguments: [Data] = []
     var throwing: Bool = false
+    
+    let system: ClusterSystem
+    
+    init(system: ClusterSystem) {
+        self.system = system
+    }
 
     public mutating func recordGenericSubstitution<T>(_ type: T.Type) throws {
         fatalError("NOT IMPLEMENTED: \(#function)")
     }
 
     public mutating func recordArgument<Value: Codable>(_ argument: RemoteCallArgument<Value>) throws {
-        let encoder = JSONEncoder()
-        let data = try encoder.encode(argument.value)
```

Comment:
the bug™ we must use the serialization infra since it sets up all the right contexts.